contracts/Removal.sol

```diff
@@ -499,6 +446,58 @@ contract Removal is
     return id.decodeRemovalIdV0();
   }
 
+  /**
+   * @dev Transfers `amount` tokens of token type `id` from `from` to `to`.
+   *
+   * Emits a {TransferSingle} event.
+   *
+   * Requirements:
+   *
+   * - Can only be called by the `Market` contract.
+   * - `to` cannot be the zero address.
+   * - If the caller is not `from`, it must have been approved to spend ``from``'s tokens via {setApprovalForAll}.
+   * - `from` must have a balance of tokens of type `id` of at least `amount`.
+   * - If `to` refers to a smart contract, it must implement {IERC1155Receiver-onERC1155Received} and return the
+   * acceptance magic value.
+   */
+  function safeTransferFrom(
+    address from,
+    address to,
+    uint256 id,
+    uint256 amount,
+    bytes memory data
+  ) public override whenNotPaused {
+    if (_msgSender() != address(_market)) {
+      revert ForbiddenTransfer();
+    }
+    super.safeTransferFrom(from, to, id, amount, data);
+  }
+
+  /**
+   * @dev Batched version of {safeTransferFrom}.
+   *
+   * Emits a {TransferBatch} event.
+   *
+   * Requirements:
```

Comment:
You'd wanna throw `#####` before this to keep the formatting the same as elsewhere